src/cascadia/TerminalSettingsModel/ActionMap.cpp

```diff
@@ -0,0 +1,553 @@
+        }
+
+        return cumulativeActions;
+    }
+
+    Windows::Foundation::Collections::IMapView<Control::KeyChord, Model::Command> ActionMap::GlobalHotkeys()
+    {
+        if (!_GlobalHotkeysCache)
+        {
+            const ActionHash actionHasher{};
+            std::set<InternalActionID> visitedActionIDs{};
+            std::unordered_map<Control::KeyChord, Model::Command, KeyChordHash, KeyChordEquality> globalHotkeys;
+            for (const auto& cmd : _GetCumulativeActions())
+            {
+                // Only populate GlobalHotkeys with actions that...
+                // (1) ShortcutAction is GlobalSummon or QuakeMode
+                const auto& actionAndArgs{ cmd.ActionAndArgs() };
+                if (actionAndArgs.Action() == ShortcutAction::GlobalSummon || actionAndArgs.Action() == ShortcutAction::QuakeMode)
+                {
+                    // (2) haven't been visited already
+                    const auto actionID{ actionHasher(actionAndArgs) };
+                    if (visitedActionIDs.find(actionID) == visitedActionIDs.end())
+                    {
+                        const auto& cmdImpl{ get_self<Command>(cmd) };
+                        for (const auto& keys : cmdImpl->KeyMappings())
+                        {
+                            // (3) haven't had that key chord added yet
+                            if (globalHotkeys.find(keys) == globalHotkeys.end())
+                            {
+                                globalHotkeys.insert({ keys, cmd });
+                            }
+                        }
+
+                        // Record that we already handled adding this action to the NameMap.
+                        visitedActionIDs.insert(actionID);
+                    }
+                }
+            }
+            _GlobalHotkeysCache = single_threaded_map<Control::KeyChord, Model::Command>(std::move(globalHotkeys));
+        }
+        return _GlobalHotkeysCache.GetView();
+    }
+
+    com_ptr<ActionMap> ActionMap::Copy() const
+    {
+        auto actionMap{ make_self<ActionMap>() };
+
+        // copy _KeyMap (KeyChord --> ID)
+        std::for_each(_KeyMap.begin(), _KeyMap.end(), [actionMap](const auto& pair) {
+            actionMap->_KeyMap.insert(pair);
+        });
+
+        // copy _ActionMap (ID --> Command)
+        for (const auto& [actionID, cmd] : _ActionMap)
+        {
+            actionMap->_ActionMap.insert({ actionID, *(get_self<Command>(cmd)->Copy()) });
+        }
+
+        // copy _NestedCommands (Name --> Command)
+        for (const auto& [name, cmd] : _NestedCommands)
+        {
+            actionMap->_NestedCommands.Insert(name, *(get_self<Command>(cmd)->Copy()));
+        }
+
+        // repeat this for each of our parents
+        FAIL_FAST_IF(_parents.size() > 1);
+        for (const auto& parent : _parents)
+        {
+            actionMap->_parents.push_back(std::move(parent->Copy()));

```

Comment:
First of all: The `std::move` should be removed. Return values in C++ are special and fall into the so called "prvalue" value category. The short story is: They behave identical to a "currently being moved" value (`T&&`) and calling `std::move` on them again is actually detrimental, as this causes the variable to potentially first being put on the stack temporarily and then being moved again into the call to `push_back` (...because `std::move` needs an address and a "prvalue" doesn't have one - you can't write `auto x = &foo();`).

BTW: The only reason C++ still has `push_back()` is because they didn't want to break the stdlib ABI. That's why they called it `emplace_back()`, but technically it's `push_back2()`, or as we at MS like to say: `PushBackEx()`.
Same for `lock_guard` vs. `scoped_lock` where the former is older and takes a single parameter, while the latter and newer version is a variadic and only exists separately to not break the ABI.
`push_back` is fine as it is in this case though. In most stdlibs, including MSVC's, it's an alias for `emplace_back` only anyways.
Same is true for other stdlib containers by the way: `insert()` is implemented in terms of `emplace()`.
Don't ask me why it's `insert_or_assign` and not `emplace_or_assign`. The C++ committee often isn't quite rational to put it mildly. 😐

Reply:
Thanks! Went through and updated the `.insert` calls in `ActionMap` so that I can be _cool_ and _hip_ 😎